client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -336,14 +355,14 @@ public void processingFailure(Exception error) {
             state.failConnection(error);
         }
     }
-    
+
     /**
-     * @see SegmentOutputStream#write(java.nio.ByteBuffer,
-     *      java.util.concurrent.CompletableFuture)
+     * @see SegmentOutputStream#write(PendingEvent)
      */
     @Override
     @Synchronized
     public void write(PendingEvent event) throws SegmentSealedException {
+        checkState(!state.isAlreadySealed(), "Segment: {} is already sealed", segmentName);
```

Comment:
Used a CompletableFuture based approach to invoke the segmentSealed call back. This approach covers the following combinations successfully without the events being lost.
1. concurrent Seg: write and SegmentSealed netty callback.
2. concurrent Seg:flush (blocking) with a SegmentSealed callback.
3. concurrent Event: Write() with SegmentSealedCallback.
4. Blocking Event: write.get() with SegmentSealedCallback.
Kindly review.